cpp/include/raft/comms/mpi_comms.hpp

```diff
@@ -24,10 +24,10 @@ namespace comms {
 
 using mpi_comms = detail::mpi_comms;
 
-inline void initialize_mpi_comms(handle_t* handle, MPI_Comm comm)
+inline void initialize_mpi_comms(handle_t* handle, MPI_Comm comm, cudaStream_t stream)
```

Comment:
Shouldn't we better use rmm::cuda_stream_view instead of cuda_stream_t?

We're using rmm::cuda_stream_view elsewhere (https://github.com/rapidsai/raft/blob/branch-22.04/cpp/include/raft/handle.hpp#L137) and better be consistent unless there is a strong reason to mix the two.